### PR TITLE
Backport to support macOS Ventura and Xcode 14 for Ruby 3.0

### PR DESCRIPTION
**Pull request by hsbt**
* https://bugs.ruby-lang.org/issues/18912
* https://bugs.ruby-lang.org/issues/19005

### CODE REVIEW

**Comment by hsbt** [comment]
I update https://github.com/ruby/ruby/commit/6898984f1cd1b0375c3da44d7832724489e0e470 for Ruby 3.0. 

@nobu Can you review the changes of `configure.ac` ?

**Comment by hsbt** [comment]
Note: nobu said "LGTM"

**Review by unak — APPROVED** [review]

File: configure.ac [hunk 1]
```diff
@@ -2884,15 +2884,26 @@ AS_IF([test "$with_dln_a_out" != yes], [
...
 	  "-multiply_defined suppress" \
+	  "-undefined dynamic_lookup" \
 	  ; do
-	  test "x${linker_flag}" = x || flag="${linker_flag}`echo ${flag} | tr ' ' ,`"
-	  RUBY_TRY_LDFLAGS([$flag], [], [flag=])
-	  AS_IF([test "x$flag" != x], [
-	    RUBY_APPEND_OPTIONS(DLDFLAGS, [$flag])
-	  ])
+      test "x${linker_flag}" = x || flag="${linker_flag}`echo ${flag} | tr ' ' ,`"
+      RUBY_TRY_LDFLAGS([$flag], [], [$flag=])
```
Comment (avmnu-sng) [inline configure.ac:2893]: @hsbt Should this not be `RUBY_TRY_LDFLAGS([$flag], [], [flag=])`? The last expression, `[flag=]` not `[$flag=]`?

Comment (hsbt) [reply configure.ac:2893]: https://github.com/ruby/ruby/blob/master/configure.ac#L3110

**Pull request merged**